src/neural/shared/activation.cc

```diff
@@ -75,44 +75,155 @@ float Activate(const float val, const ActivationFunction activation) {
   return val;
 }
 
-void Activate(const size_t len, float* data,
+void Activate(const size_t len, const float* data, const float* bias,
+              float* output, const ActivationFunction activation) {
+  if (activation == NONE) {
+    for (size_t b = 0; b < len; b++) {
+      output[b] = data[b] + bias[b];
+    }
+  } else if (activation == RELU) {
+    for (size_t b = 0; b < len; b++) {
+      float val = data[b] + bias[b];
+      output[b] = val > 0 ? val : 0;
+    }
+  } else if (activation == MISH) {
+    for (size_t b = 0; b < len; b++) {
+      float val = data[b] + bias[b];
+      auto e = expf(val);
```

Comment:
Does the mish function actually need to be manually inlined each time to get the performance? Could we at least declare it at as a separate function so that only the boiler plate is being repeated rather than the math?